File: 2011-final-value-of-variable-after-performing-operations/2011-final-value-of-variable-after-performing-operations.java
import java.util.HashMap;
class Solution {
    public int finalValueAfterOperations(String[] operations) {
        // Time Complexity: O(N)
        // Space Complexity: O(1)
        HashMap<String, Integer> mapping = new HashMap<>();
        mapping.put("++X", 1);
        mapping.put("X++", 1);
        mapping.put("--X", -1);
        mapping.put("X--", -1);

        int result = 0;
        int n = operations.length;
        for(int i=0; i<n; i++) {
            int value = mapping.get(operations[i]);
            result += value;
        }
        return result;
    }
}

File: 2011-final-value-of-variable-after-performing-operations/2011-final-value-of-variable-after-performing-operations.py
class Solution:
    def finalValueAfterOperations(self, operations: List[str]) -> int:
        # Time Complexity: O(N)
        # Space Complexity: O(1)
        mapping = {"++X": 1, "X++": 1, "--X": -1, "X--": -1}
        result = 0
        for op in operations:
            result += mapping[op]
        return result
